tools/docker/boardfarm-ci/docker-compose.yml

```diff
@@ -0,0 +1,60 @@
+version: '3'
+services:
+    # Controller and agent are run from inside the boardfarm container
+    controller:
+        image: registry.gitlab.com/prpl-foundation/prplmesh/prplmesh-runner:${CI_PIPELINE_ID}
+        privileged: true    # For the creation of the bridge to work
+        container_name: controller
+        environment:
+            - USER
+            - INSTALL_DIR=${FINAL_ROOT_DIR}/build/install
+            - ROOT_DIR=${FINAL_ROOT_DIR}
+            - CURRENT_ID
+            - CI_PIPELINE_ID
+        expose:
+            - "5000"
+            - "8002"
+        volumes:
+            - "$ROOT_DIR:${FINAL_ROOT_DIR}"
+        entrypoint: ["/root/entrypoint.sh", "/usr/bin/start-controller-agent"]
+
+    agent:
+        image: registry.gitlab.com/prpl-foundation/prplmesh/prplmesh-runner:${CI_PIPELINE_ID}
+        privileged: true
+        container_name: agent
+        environment:
+            - USER
+            - INSTALL_DIR=${FINAL_ROOT_DIR}/build/install
+            - ROOT_DIR=${FINAL_ROOT_DIR}
+              # /builds/prpl-foundation/prplMesh
+            - CURRENT_ID
+            - CI_PIPELINE_ID
+        expose:
+            - "5000"
+            - "8002"
+        volumes:
+          - "$ROOT_DIR:${FINAL_ROOT_DIR}"
+        entrypoint: ["/root/entrypoint.sh", "/usr/bin/start-agent"]
+
+    # Boardfarm image is launched from dctest.py, TARGET_DIR refers to the
+    # path inside the controller, will be referred as $ROOT_DIR inside
+    boardfarm:
+        build: .
+        privileged: true
+        container_name: boardfarm
+        environment:
+            - USER
+            - ROOT_DIR
+            - CURRENT_ID
+            - RUN_ID
+            - CI_PIPELINE_ID
+            - FINAL_ROOT_DIR
```

Comment:
Correct me if I'm wrong, but I think both FINAL_ROOT_DIR and INSTALL_DIR are not used here and can be removed.

Reply:
INSTALL_DIR environment variable is needed in the runner entrypoint.sh (tools/docker/runner/entrypoint.sh)

It is formed from FINAL_ROOT_DIR in the entries for services "controller" and "agent", but needs to be in the boardfarm image environment, as boardfarm (and not dctest.py directly) is the one launching docker-compose for those images.